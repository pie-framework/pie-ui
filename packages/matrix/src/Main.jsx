import React from 'react';
import PropTypes from 'prop-types';
import {PreviewLayout} from '@pie-lib/render-ui';
import Matrix from './Matrix';

class Main extends React.Component {
  static propTypes = {
    model: PropTypes.object,
    session: PropTypes.object,
    onSessionChange: PropTypes.func
  };

  static defaultProps = {
    model: {},
    session: {},
  };

  render() {
    const {model, onSessionChange, session} = this.props;
    return (
      <PreviewLayout>
        <Matrix
          {...model}
          session={session}
          onSessionChange={onSessionChange}
        />
      </PreviewLayout>
    );
  }
}

export default Main;
