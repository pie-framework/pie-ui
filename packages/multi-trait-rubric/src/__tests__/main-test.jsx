import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import Main from '../main';

jest.mock('@pie-lib/render-ui', () => ({
  color: {
    primaryDark: jest.fn().mockReturnValue('blue'),
    secondaryBackground: jest.fn().mockReturnValue('grey'),
    text: jest.fn().mockReturnValue('blue'),
  }
}));

describe('Main', () => {
  let wrapper, model;

  const mkWrapper = (modelOptions = {}) => {
    model = {
      halfScoring: false,
      scales: [
        {
          excludeZero: false,
          maxPoints: 1,
          scorePointsLabels: ['A', 'B'],
          traitLabel: 'Trait',
          traits: []
        }
      ],
      visible: false,
      ...modelOptions
    };

    return mount(<Main model={model}/>);
  };

  beforeEach(() => {
    wrapper = mkWrapper();
  });

  describe('snapshots', () => {
    it('not visible to students => renders empty div', () => {
      const wrapper = mkWrapper();
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('visible to students => renders table without half-scoring', () => {
      const wrapper = mkWrapper({ visible: true });
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('visible to students => renders table with half-scoring', () => {
      const wrapper = mkWrapper({ visible: true, halfScoring: true });
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('visible to students => renders empty div if no scales', () => {
      const wrapper = mkWrapper({ scales: [] });
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});
