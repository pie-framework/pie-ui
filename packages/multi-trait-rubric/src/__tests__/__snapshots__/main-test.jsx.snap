// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main snapshots not visible to students => renders empty div 1`] = `
<Main
  model={
    Object {
      "halfScoring": false,
      "scales": Array [
        Object {
          "excludeZero": false,
          "maxPoints": 1,
          "scorePointsLabels": Array [
            "A",
            "B",
          ],
          "traitLabel": "Trait",
          "traits": Array [],
        },
      ],
      "visibleToStudent": false,
    }
  }
>
  <div />
</Main>
`;

exports[`Main snapshots visible to students => renders empty div if no scales 1`] = `
<Main
  model={
    Object {
      "halfScoring": false,
      "scales": Array [],
      "visibleToStudent": false,
    }
  }
>
  <div />
</Main>
`;

exports[`Main snapshots visible to students => renders table with half-scoring 1`] = `
<Main
  model={
    Object {
      "halfScoring": true,
      "scales": Array [
        Object {
          "excludeZero": false,
          "maxPoints": 1,
          "scorePointsLabels": Array [
            "A",
            "B",
          ],
          "traitLabel": "Trait",
          "traits": Array [],
        },
      ],
      "visibleToStudent": true,
    }
  }
>
  <div>
    <p>
      * Half-point or in-between scores are permitted under this rubric.
    </p>
    <WithStyles(Scale)
      key="scale_0"
      scale={
        Object {
          "excludeZero": false,
          "maxPoints": 1,
          "scorePointsLabels": Array [
            "A",
            "B",
          ],
          "traitLabel": "Trait",
          "traits": Array [],
        }
      }
      scaleIndex={0}
    >
      <Scale
        classes={
          Object {
            "scorePointHeader": "Scale-scorePointHeader-2",
            "table": "Scale-table-1",
          }
        }
        scale={
          Object {
            "excludeZero": false,
            "maxPoints": 1,
            "scorePointsLabels": Array [
              "A",
              "B",
            ],
            "traitLabel": "Trait",
            "traits": Array [],
          }
        }
        scaleIndex={0}
      >
        <table
          className="Scale-table-1"
          key="scale-0"
        >
          <thead>
            <tr>
              <th>
                <div
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "Trait",
                    }
                  }
                />
              </th>
              <th
                key="table-header-0"
              >
                <table
                  className="Scale-scorePointHeader-2"
                >
                  <thead>
                    <tr>
                      <td>
                        1
                      </td>
                    </tr>
                  </thead>
                  <tbody>
                    <tr>
                      <td>
                        <div
                          dangerouslySetInnerHTML={
                            Object {
                              "__html": "B",
                            }
                          }
                        />
                      </td>
                    </tr>
                  </tbody>
                </table>
              </th>
              <th
                key="table-header-1"
              >
                <table
                  className="Scale-scorePointHeader-2"
                >
                  <thead>
                    <tr>
                      <td>
                        0
                      </td>
                    </tr>
                  </thead>
                  <tbody>
                    <tr>
                      <td>
                        <div
                          dangerouslySetInnerHTML={
                            Object {
                              "__html": "A",
                            }
                          }
                        />
                      </td>
                    </tr>
                  </tbody>
                </table>
              </th>
            </tr>
          </thead>
          <tbody />
        </table>
      </Scale>
    </WithStyles(Scale)>
  </div>
</Main>
`;

exports[`Main snapshots visible to students => renders table without half-scoring 1`] = `
<Main
  model={
    Object {
      "halfScoring": false,
      "scales": Array [
        Object {
          "excludeZero": false,
          "maxPoints": 1,
          "scorePointsLabels": Array [
            "A",
            "B",
          ],
          "traitLabel": "Trait",
          "traits": Array [],
        },
      ],
      "visibleToStudent": true,
    }
  }
>
  <div>
    <WithStyles(Scale)
      key="scale_0"
      scale={
        Object {
          "excludeZero": false,
          "maxPoints": 1,
          "scorePointsLabels": Array [
            "A",
            "B",
          ],
          "traitLabel": "Trait",
          "traits": Array [],
        }
      }
      scaleIndex={0}
    >
      <Scale
        classes={
          Object {
            "scorePointHeader": "Scale-scorePointHeader-2",
            "table": "Scale-table-1",
          }
        }
        scale={
          Object {
            "excludeZero": false,
            "maxPoints": 1,
            "scorePointsLabels": Array [
              "A",
              "B",
            ],
            "traitLabel": "Trait",
            "traits": Array [],
          }
        }
        scaleIndex={0}
      >
        <table
          className="Scale-table-1"
          key="scale-0"
        >
          <thead>
            <tr>
              <th>
                <div
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "Trait",
                    }
                  }
                />
              </th>
              <th
                key="table-header-0"
              >
                <table
                  className="Scale-scorePointHeader-2"
                >
                  <thead>
                    <tr>
                      <td>
                        1
                      </td>
                    </tr>
                  </thead>
                  <tbody>
                    <tr>
                      <td>
                        <div
                          dangerouslySetInnerHTML={
                            Object {
                              "__html": "B",
                            }
                          }
                        />
                      </td>
                    </tr>
                  </tbody>
                </table>
              </th>
              <th
                key="table-header-1"
              >
                <table
                  className="Scale-scorePointHeader-2"
                >
                  <thead>
                    <tr>
                      <td>
                        0
                      </td>
                    </tr>
                  </thead>
                  <tbody>
                    <tr>
                      <td>
                        <div
                          dangerouslySetInnerHTML={
                            Object {
                              "__html": "A",
                            }
                          }
                        />
                      </td>
                    </tr>
                  </tbody>
                </table>
              </th>
            </tr>
          </thead>
          <tbody />
        </table>
      </Scale>
    </WithStyles(Scale)>
  </div>
</Main>
`;
